src/components/AddEditMovie/AddEditMovie.jsx

```diff
@@ -1,93 +1,171 @@
+                }
+
+                if ((formik.touched.release_date && !isValidDate(formik.values.release_date)) || (isSubmitButton && !isValidDate(formik.values.release_date))) {
+                    errors.release_date = "*Not a valid Release Date. Please enter in YYYY-MM-DD format.";
+                }
+
+                if ((formik.touched.poster_path && !formik.values.poster_path) || (isSubmitButton && !formik.values.poster_path)) {
+                    errors.poster_path = "*Movie URL is a required field";
+                } else if ((formik.touched.poster_path && !isValidUrl(formik.values.poster_path)) || (isSubmitButton && !isValidUrl(formik.values.poster_path))) {
+                    errors.poster_path = "*Not a valid Movie URL";
+                }
+
+                if ((formik.touched.overview && !formik.values.overview) || (isSubmitButton && !formik.values.overview)) {
+                    errors.overview = "*Overview is a required field";
+                }
+
+                if ((formik.touched.genres && !formik.values.genres) || (isSubmitButton && !formik.values.genres)) {
+                    errors.genres = "*Genres is a required field";
+                }
+
+                if ((formik.touched.runtime && !formik.values.runtime) || (isSubmitButton && !formik.values.runtime)) {
+                    errors.runtime = "*Runtime is a required field";
+                }
+                else if ((formik.touched.runtime && !isNumber(formik.values.runtime)) || (isSubmitButton && !isNumber(formik.values.runtime))) {
+                    errors.runtime = "Runtime accepts only numbers";
+                }
+
+                if (isSubmitButton) {
+                    boxRef.current.scrollIntoView();
+                    isSubmitButton = false;
+                }
+
+                return errors;
+            },
+            onSubmit(values) {
+                values.genres = (values.genres).split(',');
+                values.runtime = Number(values.runtime);
+                console.log("Form values: ", values);
+                props.onAddMovieSubmitAction(values);
+            }
+        })
 
     return (
         (props.editMovieSelection !== undefined) ?
             <>
                 <div id="myModal" className="modal">
-
-
                     <div className="modal-content">
                         <div className="modal-header">
                             <span className="close" onClick={() => props.onCloseAction()}>&times;</span>
                         </div>
-
                         <div className="modal-body">
                             <label className="modal-header-lbl-properties">EDIT MOVIE</label>
                             <br /><br />
-                            <form id="form-edit-movie">
+                            <form id="form-edit-movie" onSubmit={formik.handleSubmit}>
                                 <p className="modal-body-titles">MOVIE ID</p>
```

Comment:
Usually, you must not display **ID** of the element in the Edit form. If you skip displaying ID, you can use the same JSX for Edit and Add forms.

Reply:
too much duplication in this component